core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -539,6 +543,176 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
+
+  /**
+   * Creates Resource znode with the given acls
+   * @param resource  resource object
+   * @param aclsSet the acls to sets on the resource znode
+   * @return CreateResponse
+   */
+  def createAclForResource(resource: Resource, aclsSet: Set[Acl]): CreateResponse = {
+    val path = ResourceZNode.path(resource)
+    val createRequest = CreateRequest(path, ResourceZNode.encode(aclsSet), acls(path), CreateMode.PERSISTENT)
+    retryRequestUntilConnected(createRequest)
+  }
+
+  /**
+   * Creates Acl change notification message
+   * @param resource resource name
+   * @return CreateResponse
+   */
+  def createAclChangeNotification(resource: String): CreateResponse = {
+    val path = AclChangeNotificationSequenceZNode.path
+    val createRequest = CreateRequest(path, AclChangeNotificationSequenceZNode.encode(resource), acls(path), CreateMode.PERSISTENT_SEQUENTIAL)
+    retryRequestUntilConnected(createRequest)
+  }
+
+  /**
+   * Gets the resource types
+   */
+  def getResourceTypes(): Seq[String] = {
+    val getChildrenResponse = retryRequestUntilConnected(GetChildrenRequest(AclZNode.path))
+    if (getChildrenResponse.resultCode == Code.OK) {
+      getChildrenResponse.children
+    } else if (getChildrenResponse.resultCode == Code.NONODE) {
+      Seq.empty
+    } else {
+      throw getChildrenResponse.resultException.get
+    }
+  }
+
+  /**
+   * Gets the resource names for a give resource type
+   * @param resourceType
+   * @return list of resource names
+   */
+  def getResourceNames(resourceType: String): Seq[String] = {
+    val getChildrenResponse = retryRequestUntilConnected(GetChildrenRequest(ResourceTypeZNode.path(resourceType)))
+    if (getChildrenResponse.resultCode == Code.OK) {
+      getChildrenResponse.children
+    } else if (getChildrenResponse.resultCode == Code.NONODE) {
+      Seq.empty
+    } else {
+      throw getChildrenResponse.resultException.get
+    }
+  }
+
+  /**
+   * Deletes the given Resource node
+   */
+  def deleteResource(resource: Resource): Boolean = {
+    deleteRecursive(ResourceZNode.path(resource))
+    true
+  }
+
+  /**
+   * Conditional delete the resource node
+   * @param resource
+   * @param expectedVersion
+   * @return return true if it succeeds, false otherwise (the current version is not the expected version)
+   */
+  def conditionalDelete(resource: Resource, expectedVersion: Int): Boolean = {
+    val deleteRequest = DeleteRequest(ResourceZNode.path(resource), expectedVersion)
+    val deleteResponse = retryRequestUntilConnected(deleteRequest)
+    if (deleteResponse.resultCode == Code.OK || deleteResponse.resultCode == Code.NONODE) {
+      true
+    } else if (deleteResponse.resultCode == Code.BADVERSION) {
+      false
+    } else {
+      throw deleteResponse.resultException.get
+    }
+  }
+
+  /**
+   * Gets all the child nodes at a given zk node path
+   * @param path
+   * @return list of child node names
+   */
+  def getChildren(path : String): Seq[String] = {
```

Comment:
Could other methods like getResourceTypes() and getResourceNames(), etc that creates GetChildrenRequest directly just call this method?

Reply:
Could you make this private?